data/projects/editor_support.md

```diff
@@ -1,31 +1,68 @@
+plugins.
 
-We consider any bit of code *supported* only if we have regular continuous
-integration and tests running to ensure it's functionality.
+The goal is to develop an efficient strategy that allows coala to maintain a 
+number of plugins for different editors. This would not only benefit coala
+attracting more developers as the plugins across various platforms will be 
+widely used, but it would also benefit the end-users by giving them a working
+linter plugin across their favourite editors which are easily maintained and
+tested regularly with the latest build of coala.
 
-coala features quite some editor plugins already, none of which are *supported*
-according to the above definition. The goal of this project is to plan,
-introduce and configure a testing routine that allows coala to ensure the
-functioning of it's editor plugins at all times.
-A viable approach might be to set up weekly tests for the current editor
-plugins. For some editors like sublime and gedit, running tests is not easy and
-there is no api to do it. In such cases we  would want to use dogtail
+Details (for testing): 
+
+For sublime, we'll be using the 
+[unittest plugin](https://github.com/randy3k/UnitTesting)
+which is a plugin by itself. 
+For atom, we have the apm testing architecture
+which does not require a GUI.
+For VI and emacs, we will most likely use a testing framework such as 
+[Vimcasts](https://github.com/kana/vim-vspec).
+For some editors like gedit, running tests is not easy and
+there is no api to do it. In such cases we  would want to use 
+[dogtail](https://fedorahosted.org/dogtail/)
 (a GUI tester) or a similar tool to do this.
 
+Details (for maintenance):
+
+We will have to explore the possibility of having a single git repository for 
+atom, sublime-text and vim as it will ultimately be easier to maintain in the
+long run and will have a proper CI, which can be easily managed. Issues will 
+be raised more frequently as a result of this.
+
+Details (for documentation):
+
+Documentation which isn't available at the moment will have to describe the 
+usage details of existing plugins, we can have asciicinemas or gifs of the
+entire process from installation to examples of the plugins in action.
+Documentation can also be created for future plugins, and on the basis of the
+tests which will be written for the existing ones, it'll be easier to test
+the future plugins. This will result in a clear cut path of creating plugins
+for coala.
+
 
 #### MILESTONES
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
 * A cEP for the testing, documentation and maintaining strategy has been merged.
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
 * The testing strategy has been set up for coala.
 * At least one of coala's editor plugins is continually tested.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
+
+* Documentation and tutorials showing the usage of existing plugins is
+available.
+* Maintaining strategy is placed into effect.
```

Comment:
Well for this I was thinking 'maintaining strategy' is just the decision of whether or not the single repo thing is gonna happen. Also, in order to fit in with phase-3(at least 3), the student would have had to begun work on the testing for the second plugin aswell. If the repo thing is gonna happen, then the second plugin would go there accordingly. Else, separate. Hence it stays true to the objective of whatever the maintaining strategy is, it'd be placed into effect. 
An addition that could be made here is the 'second plugin is tested' but I felt thatd be repetitive with the first and third milestones and its sort of a given.